centreon/www/front_src/src/ApiTokens/TokenCreation/TokenCreationDialog.tsx

```diff
@@ -1,16 +1,33 @@
+import { ReactNode } from 'react';
+
 import { Formik } from 'formik';
 import { useTranslation } from 'react-i18next';
 import { number, object, string } from 'yup';
 
+import { ResponseError } from '@centreon/ui';
+
 import { CreateTokenFormValues } from '../TokenListing/models';
 import { labelFieldRequired } from '../translatedLabels';
+import useRefetch from '../useRefetch';
 
-import FormCreation from './Form';
+import { CreatedToken } from './models';
 import useCreateToken from './useCreateToken';
 
-const TokenCreationDialog = (): JSX.Element => {
+interface Parameters {
+  data?: ResponseError | CreatedToken | undefined;
+  isMutating: boolean;
+  isRefetching: boolean;
+}
+
+interface Props {
+  renderFormCreation: (params: Parameters) => ReactNode;
+}
+
+const TokenCreationDialog = ({ renderFormCreation }: Props): JSX.Element => {
```

Comment:
Be careful about passing function as prop

Reply:
to avoid props drilling :running:

Reply:
You are doing props drilling atm.